transport/http/handler.go

```diff
@@ -267,7 +277,7 @@ func (rw *responseWriter) Write(s []byte) (int, error) {
 }
 
 func (rw *responseWriter) AddHeaders(h transport.Headers) {
-	applicationHeaders.ToHTTPHeaders(h, rw.w.Header())
+	_, rw.err = applicationHeaders.ToHTTPHeaders(h, rw.w.Header(), rw.edgeMetrics)
```

Comment:
should you use https://pkg.go.dev/go.uber.org/multierr to combine the errors here instead of override every time?

Reply:
AddHeaders is called only once, so that's why we can just set err